src/sentry/static/sentry/app/components/bases/pluginComponentBase.tsx

```diff
@@ -11,17 +11,28 @@ import {
 } from 'app/actionCreators/indicator';
 import {t} from 'app/locale';
 
-const callbackWithArgs = function (callback, ...args) {
+const callbackWithArgs = function (context: any, callback, ...args) {
   if (isFunction(callback)) {
-    callback = callback.bind(this, ...args);
+    callback = callback.bind(context, ...args);
   } else {
     callback = null;
   }
   return callback;
 };
 
-class PluginComponentBase extends React.Component {
-  constructor(props, context) {
+type GenericFieldProps = Parameters<typeof GenericField>[0];
+
+type Props = {
+  config?: GenericFieldProps['config'];
+} & GenericFieldProps;
```

Comment:
I believe you forgot to omit the config here 

```suggestion
} & Omit<GenericFieldProps, 'config'>;
```